website/content/docs/connect/dataplane/telemetry.mdx

```diff
@@ -0,0 +1,92 @@
+- Pass the
+  [`-telemetry-use-central-config`](/docs/connect/dataplane/consul-dataplane#telemetry-use-central-config)
+  flag to `consul-dataplane`.
+
+## Prometheus Metrics Merging
+
+When Prometheus metrics are used, Consul Dataplane configures Envoy to serve merged metrics through
+a single endpoint. Metrics from the following sources are collected and merged:
+
+- The Envoy process managed by Consul Dataplane
+- Consul Dataplane itself
+- (optionally) Your service that is running alongside Consul Dataplane
+
+If your service supports Prometheus metrics, set the
+[`-telemetry-prom-service-metrics-url`](/docs/connect/dataplane/consul-dataplane#telemetry-prom-service-metrics-url)
+option to the URL where your service exposes Prometheus metrics in order to include metrics for your
+service in the merged metrics.
+
+## Additional Prometheus Options
+
+To configure TLS for communicating to the Prometheus server, use the following flags:
+
+- [`-telemetry-prom-ca-certs-path`](/docs/connect/dataplane/consul-dataplane#telemetry-prom-ca-certs-path)
+- [`-telemetry-prom-cert-file`](/docs/connect/dataplane/consul-dataplane#telemetry-prom-cert-file)
+- [`-telemetry-prom-key-file`](/docs/connect/dataplane/consul-dataplane#telemetry-prom-key-file)
+
+To change the port Consul Dataplane listens on for metrics merging, use the
+[`-telemetry-prom-merge-port`](/docs/connect/dataplane/consul-dataplane#telemetry-prom-merge-port)
+flag. You should set this if your service needs to listen on the default metrics merging port,
+20100, in order to avoid a port conflict.
+
+To change the duration Prometheus metrics are retained for, use the
+[`-telemetry-prom-retention-time`](/docs/connect/dataplane/consul-dataplane#telemetry-prom-retention-time)
+flag. Refer to
+[`prometheus_retention_time`](/docs/agent/config/config-files#telemetry-prometheus_retention_time)
+for details on how this field should be set.
+
+To change the URL path where Envoy serves Prometheus metrics, use the
+[`-telemetry-prom-scrape-path`](/docs/connect/dataplane/consul-dataplane#telemetry-prom-scrape-path)
+flag.
+
+</Tab>
+
+</Tabs>
+
+# Metrics
+
+Consul Dataplane supports the following metrics:
+
+| Metric Name                                  | Description                                                                                                                                                                               | Unit             | Type    |
+| :------------------------------------------- | :---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | :--------------- | :------ |
+| `consul_dataplane.connect_duration`          | Measures the time `consul-dataplane` spends connecting to a server. This includes everything from discovery of Consul server addresses up to opening an xDS stream to the Consul servers. | ms               | timer   |
```

Comment:
```suggestion
| `consul_dataplane.connect_duration`          | Measures the time `consul-dataplane` spends connecting to a server.  | ms               | timer   |
```
Might suffice? The time to open the xDS stream is not really computed, yea?

Reply:
That's true. This doesn't include opening the xDS stream. But I do want to clarify that it is measuring multiple steps (i.e. discovery, connection, login, fetching dataplane features), and not just a TCP connection time.